kartothek/io_components/metapartition.py

```diff
@@ -1229,8 +1229,11 @@ def build_indices(self, columns):
             possible_values = set()
             col_in_partition = False
             for df in self.data.values():
+
                 if col in df:
-                    possible_values = possible_values | set(df[col].dropna().unique())
+                    if df[col].hasnans:
```

Comment:
@xhochy you might have an opinion about this as well

Reply:
Sounds legit to disallow NULLs here. There is a possibility to encode nulls so that other tools also understand that but that turned out to be a headache for them, so raising is best.

Reply:
This is not about partitioning but indexing. Therefore, there is no encoding since ktk is the only user of these values.

partitioning raises already, the question is whether or not secondary indices should simply drop these. I see the possibility to implement this properly with `pd.NA`once this really works but until then this is a bit of no-mans land. So far, we silently dropped it

Reply:
Ah, I first defer then to @mlondschien before I make new comment.